server.go

```diff
@@ -45,10 +47,10 @@ func main() {
 	log.Printf("HTTP Write Timeout: %s\n", cfg.WriteTimeout)
 
 	bootstrapHandlers := bootTypes.FaaSHandlers{
-		FunctionProxy:  handlers.FunctionProxy(true, dockerClient),
 		DeleteHandler:  handlers.DeleteHandler(dockerClient),
 		DeployHandler:  handlers.DeployHandler(dockerClient, maxRestarts, restartDelay),
 		FunctionReader: handlers.FunctionReader(true, dockerClient),
+		FunctionProxy:  proxy.NewHandlerFunc(cfg.ReadTimeout, handlers.NewFunctionLookup(dockerClient, cfg.DNSRoundRobin)),
```

Comment:
Can we break out `handlers.NewFunctionLookup` so it's created above? Also is there any possibility of `handlers.NewFunctionLookup` returning an error, or no?

Reply:
there should be no errors, all it does is create a handler instance, any errors would only occur when creating the `dockerClient` instance.